pipecat-starters/smallwebrtc/client/src/app.ts

```diff
@@ -0,0 +1,205 @@
+        onTrackStarted: (
+          track: MediaStreamTrack,
+          participant?: Participant
+        ) => {
+          if (!participant?.local) {
+            this.onBotTrackStarted(track);
+          }
+        },
+        onServerMessage: (msg: unknown) => {
+          this.log(`Server message: ${msg}`);
+        },
+      },
+    };
+    this.pcClient = new PipecatClient(opts);
+    // @ts-ignore
+    window.webapp = this;
+    // @ts-ignore
+    window.client = this.pcClient; // Expose client for debugging
+    this.smallWebRTCTransport = this.pcClient.transport as SmallWebRTCTransport;
+  }
+
+  private setupDOMElements(): void {
+    this.connectBtn = document.getElementById(
+      'connect-btn'
+    ) as HTMLButtonElement;
+    this.disconnectBtn = document.getElementById(
+      'disconnect-btn'
+    ) as HTMLButtonElement;
+    this.debugLog = document.getElementById('debug-log');
+    this.statusSpan = document.getElementById('connection-status');
+    this.botAudioElement = document.getElementById('bot-audio') as HTMLAudioElement;
+  }
+
+  private setupDOMEventListeners(): void {
+    this.connectBtn.addEventListener('click', () => this.start());
+    this.disconnectBtn.addEventListener('click', () => this.stop());
+  }
+
+  private log(message: string): void {
+    if (!this.debugLog) return;
+    const entry = document.createElement('div');
+    entry.textContent = `${new Date().toISOString()} - ${message}`;
+    if (message.startsWith('User: ')) {
+      entry.style.color = '#2196F3';
+    } else if (message.startsWith('Bot: ')) {
+      entry.style.color = '#4CAF50';
+    }
+    this.debugLog.appendChild(entry);
+    this.debugLog.scrollTop = this.debugLog.scrollHeight;
+  }
+
+  private clearAllLogs() {
+    this.debugLog!.innerText = '';
+  }
+
+  private updateStatus(status: string): void {
+    if (this.statusSpan) {
+      this.statusSpan.textContent = status;
+    }
+    this.log(`Status: ${status}`);
+  }
+
+  private onConnectedHandler() {
+    this.updateStatus('Connected');
+    if (this.connectBtn) this.connectBtn.disabled = true;
+    if (this.disconnectBtn) this.disconnectBtn.disabled = false;
+  }
+
+  private onDisconnectedHandler() {
+    this.updateStatus('Disconnected');
+    if (this.connectBtn) this.connectBtn.disabled = false;
+    if (this.disconnectBtn) this.disconnectBtn.disabled = true;
+  }
+
+  private onBotTrackStarted(track: MediaStreamTrack) {
+    if (track.kind === 'audio') {
+      this.botAudioElement.srcObject = new MediaStream([track]);
+    }
+  }
+
+  private async start(): Promise<void> {
+    this.clearAllLogs();
+    await this.pcClient.initDevices()
+    this.connectBtn.disabled = true;
+    try {
+      this.updateStatus('Starting the bot');
+      const headers = new Headers();
+      headers.append("Authorization", `Bearer ${this.apiKey}`);
+      const startBotResult = await this.pcClient.startBot({
+          endpoint: this.startUrl,
+          headers: headers
+      })
```

Comment:
Need to add a comment that it is not supposed to be used like this in production. It is just for making local tests easier.